spikeinterface/extractors/stream_ibl.py

```diff
@@ -0,0 +1,204 @@
+        In ONE, this is sometimes referred to as the 'eid'.
+        When doing a session lookup such as
+
+            sessions = one.alyx.rest('sessions', 'list', tag='2022_Q2_IBL_et_al_RepeatedSite')
+
+        each returned value in `sessions` refers to it as the 'id'.
+    stream_name : str
+        The name of the stream to load for the session.
+        These can be retrieved from calling `StreamingIblExtractor.get_stream_names(session="<your session ID>")`.
+    load_sync_channels : bool, default: false
+        Load or not the last channel (sync).
+        If not then the probe is loaded.
+    cache_folder : str, optional
+        The location to temporarily store chunks of data during streaming.
+        The default uses the folder designated by ONE.alyx._par.CACHE_DIR / "cache".
+    remove_cached : bool, default: true
+        Whether or not to remove data from the cache after it is read.
+        If you expect to reuse fetched data many times, and have the disk space available, it is recommended to set this to False.
+
+    Returns
+    -------
+    recording : StreamingIblExtractor
+        The recording extractor which allows access to the traces.
+    """
+
+    extractor_name = "StreamingIbl"
+    has_default_locations = True
+    installed = HAVE_BRAINBOX
+    mode = "folder"
+    installation_mesg = (
+        "To use the StreamingIblExtractor, install ONE-api and ibllib: \n\n pip install ONE-api\npip install ibllib\n"
+    )
+    name = "stream_ibl"
+
+    @classmethod
+    def get_stream_names(cls, session: str) -> List[str]:
+        """
+        Convenient retrieval of available stream names.
+
+        Parameters
+        ----------
+        session : str
+          DESCRIPTION.
+
+        Returns
+        -------
+        stream_names : list of str
+            List of stream names as expected by the `stream_name` argument for the class initialization.
+        """
+        from one.api import ONE
+
+        one = ONE(base_url="https://openalyx.internationalbrainlab.org", password="international", silent=True)
+
+        dataset_contents = one.list_datasets(eid=session, collection="raw_ephys_data/*")
+        raw_contents = [dataset_content for dataset_content in dataset_contents if not dataset_content.endswith(".npy")]
+        probe_labels = set([raw_content.split("/")[1] for raw_content in raw_contents])
+
+        stream_names = list()
+        for probe_label in probe_labels:
+            raw_suffixes_by_probe = set(
+                [Path(raw_content).suffixes[-2] for raw_content in raw_contents if "probe00" in raw_content]
+            )
+            if ".ap" in raw_suffixes_by_probe:
+                stream_names.append(probe_label + ".ap")
+            if ".lf" in raw_suffixes_by_probe:
+                stream_names.append(probe_label + ".lf")
+
+        return stream_names
+
+    def __init__(
+        self,
+        session: str,
+        stream_name: str,
+        load_sync_channel: bool = False,
```

Comment:
[A note from the IBL docs](https://int-brain-lab.github.io/iblenv/notebooks_external/loading_raw_ephys_data.html#Option-1:-Stream-snippets-of-raw-ephys-data) indicates this might not be exactly the same in behavior as it is for the SpikeGlxRecordingExtractor (where it's just that one extra single channel from the ap.bin or lf.bin file), they seem to indicate it could be multiple AUX channels leading into a NIDQ/DAQ but I have yet to find example files where this is the case

Reply:
Then it probably makes sense to call it `load_sync_channels` and to adopt @weiglszonja 's suggestion of using `file_streamer.nsync`. What do you think @CodyCBakerPhD ?

Reply:
See my other notes on the matter - despite that note I left here about _their_ documentation, it appears as if the argument always follows the same behavior as it does on every other SpikeGlxExtractor, where there is only a single extra channel for each band

Reply:
I'm running a holistic scan to confirm then we can resolve the conversations

```python
from spikeinterface.extractors import StreamingIblExtractor
from tqdm import tqdm_notebook
from one.api import ONE

one = ONE(base_url='https://openalyx.internationalbrainlab.org', password='international', silent=True)

all_sessions = list(one.alyx.rest(url="sessions", action="list"))

all_sessions_nsync = dict()
for session in tqdm_notebook(all_sessions):
    session_id = session["id"]
    for stream_name in StreamingIblExtractor.get_stream_names(session=session_id):
        recording = StreamingIblExtractor(session=session_id, stream_name=stream_name)
        all_sessions_nsync.update({session_id: recording._file_streamer.nsync})
```

ETA 5 hours

Reply:
:) nice!

Reply:
Well, that ETA (according to tqdm) is probably off by quite a bit, its starting with the sessions that have ecephys but only a small percentage of the total number of sessions have any so we'll see

Reply:
Yeah, only took like 15 minutes.

Confirmed the usage of `nsync`

```python
set(all_sessions_nsync.values())
>>> {1}
```

Reply:
ok then! Then let's merge as is. We can always make changes later if things change